python/ray/_private/runtime_env/conda_utils.py

```diff
@@ -92,18 +92,24 @@ def create_conda_env(conda_yaml_file: str,
             f"Conda executable by setting the {RAY_CONDA_HOME} "
             "environment variable to the path of the Conda executable.")
 
+    _, stdout, _ = exec_cmd([conda_path, "env", "list", "--json"])
+    envs = json.loads(stdout)["envs"]
+
+    if prefix in envs:
+        logger.info(f"Conda environment {prefix} already exists.")
+        return
+
     create_cmd = [
         conda_path, "env", "create", "--file", conda_yaml_file, "--prefix",
         prefix
     ]
 
-    if create_cmd is not None:
-        logger.info(f"Creating conda environment {prefix}")
-        exit_code, output = exec_cmd_stream_to_logger(create_cmd, logger)
-        if exit_code != 0:
-            shutil.rmtree(prefix)
-            raise RuntimeError(
-                f"Failed to install conda environment {prefix}:\n{output}")
+    logger.info(f"Creating conda environment {prefix}")
+    exit_code, output = exec_cmd_stream_to_logger(create_cmd, logger)
```

Comment:
Sorry for the unrelated message; But does it block the main thread of agents? This seems to be dangerous if that's the case

Reply:
Good question, it's running in a separate thread: https://github.com/ray-project/ray/blob/master/dashboard/modules/runtime_env/runtime_env_agent.py#L148

Reply:
good to know! And glad it is already running in a separate thread :)

Reply:
```
Changed in version 3.5: If max_workers is None or not given, it will default to the number of processors on the machine, multiplied by 5, assuming that ThreadPoolExecutor is often used to
```

So, we start num_cpus * 5 default executors, which I think is pretty good for runtime env (IO heavy)